aspnetcore/fundamentals/primitives/change-tokens.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Detect changes with change tokens in ASP.NET Core
+author: guardrex
+description: Learn how to use change tokens to track changes.
+manager: wpickett
+ms.author: riande
+ms.date: 11/10/2017
+ms.devlang: csharp
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: fundamentals/primitives/change-tokens
+---
+# Detect changes with change tokens in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+A *change token* is a general-purpose, low-level building block used to track changes.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/primitives/change-tokens/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## IChangeToken interface
+
+The public interface [IChangeToken](/dotnet/api/microsoft.extensions.primitives.ichangetoken) propagates notifications that a change has occurred. `IChangeToken` resides in the [Microsoft.Extensions.Primitives](/dotnet/api/microsoft.extensions.primitives) namespace. For apps that don't use the [Microsoft.AspNetCore.All](https://www.nuget.org/packages/Microsoft.AspNetCore.All/) metapackage, reference the [Microsoft.Extensions.Primitives](https://www.nuget.org/packages/Microsoft.Extensions.Primitives/) NuGet package in the project file.
+
+The interface has two properties:
+
+* [ActiveChangedCallbacks](/dotnet/api/microsoft.extensions.primitives.ichangetoken.activechangecallbacks) indicate if the token proactively raises callbacks. If `ActiveChangedCallbacks` is set to `false`, a callback is never called, and the app must poll `HasChanged` for changes. It's also possible a token will never be cancelled when `ActiveChangedCallbacks` is set to `false`.
```

Comment:
"It's also possible for a token to never be cancelled if no changes occur or the underlying change listener is cleaned up." This is unrelated to ActiveChangeCallbacks